src/Controller/ArticlesController.php

```diff
@@ -881,33 +901,22 @@ private function findFigures(PromiseInterface $item) : PromiseSequence
         }));
     }
 
-    private function generateTextPath(ArticleHistory $history, int $forVersion = null, string $fragment = null) : string
+    private function generatePath(ArticleHistory $history, int $forVersion = null, string $subRoute = null, string $fragment = null) : string
     {
-        $currentVersion = $history->getVersions()[count($history->getVersions()) - 1];
-
-        if (null === $forVersion) {
-            $forVersion = $currentVersion->getVersion();
-        }
-
-        if ($forVersion === $currentVersion->getVersion()) {
-            return $this->get('router')->generate('article', [$currentVersion, '_fragment' => $fragment]);
+        if ($subRoute) {
+            $subRoute = "-{$subRoute}";
         }
```

Comment:
bugs me a bit overwriting a parameter and changing its type, could be 
```
$route = 'article';
if ($subroute) {
    $route .= "-{$subroute}";
}
```
?

Reply:
Not changing the type.

We'd need a second variable for the `article-version` routes.

Reply:
as in "null can't be concatenated or substituted into a string", but it's really a minor point

Reply:
It won't be `null` at this point.